plugin-unity-agent/src/test/kotlin/jetbrains/buildServer/unity/UnityVersionsDetectorTest.kt

```diff
@@ -25,7 +25,7 @@ class UnityVersionsDetectorTest {
     @Test
     fun testFindInstallations() {
         val detector = when {
-            SystemInfo.isWindows -> WindowsUnityDetector()
+            SystemInfo.isWindows -> WindowsUnityDetector(PEProductVersionDetector())
```

Comment:
What do you think about adding broken (like in an original topic) binary file to check this logic in tests?

Reply:
For sure, it makes sense to add some "edge case" binary. The problem is there are not many lightweight executables with specific structures. I'll investigate it later, as we don't have tests for that anyway